src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -29,17 +28,28 @@ internal KestrelConfigurationLoader(KestrelServerOptions options, IConfiguration
         {
             Options = options ?? throw new ArgumentNullException(nameof(options));
             Configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
-            ConfigurationReader = new ConfigurationReader(Configuration);
         }
 
         public KestrelServerOptions Options { get; }
-        public IConfiguration Configuration { get; }
-        internal ConfigurationReader ConfigurationReader { get; }
+        public IConfiguration Configuration { get; internal set; }
+
+        /// <summary>
+        /// If <see langword="true"/>, Kestrel will reload endpoints defined in the "Endpoints" section of the <see cref="Configuration"/>, if they are changed.
+        /// Kestrel will bind and unbind to endpoints as necessary.
+        /// </summary>
+        /// <remarks>Code-defined endpoints defined it code will not be reloaded even if the default certificate or protocol used by endpoint changed.</remarks>
+        public bool ReloadOnChange { get; } = true;
```

Comment:
This is the only new public API in this PR.

Reply:
Should this public API instead be on `KestrelServerOptions.Configure(IConfiguration, bool reloadOnChange)`? Nobody interacts directly with KestrelConfigurationLoader.

Get only? Can't be changed?

Reply:
We still default to true for the existing overload, right?

Reply:
This is new functionality, is there any chance of it being disruptive? E.g. what happens if someone saves the appsettings.json file with a syntax error? Does the config change event still fire if the file can't be parsed?